### DISCUSSION TITLE
How can I ensure WinGet properly detects my software (as a dev/publisher)

### DISCUSSION THREAD

**Discussion by marticliment** [microsoft/winget-cli#4827] [Q&A] [answered]
_This has likely been discussed somewhere else, but I have not been able to find where._

I have realized that WinGet does not always detect that an user has (externally) installed a program available to download from WinGet, or the program gets misdetected for the pre-release version (when two listings are published on WinGet, one for stable and other for beta). 
This can be frustrating for end users, as they see how their app is getting updated to the pre-release versions (likely unstable), when they did install the stable release.

Is there any way in which the developer/publisher can ensure that the program will be properly detected by WinGet? Something like adding a registry value somewhere on install time, or adding the _apps and features id_ id on the manifest, etc.

If not, perhaps such a feature would be a great addition, and would really improve reliability for certain packages, and improve the overall experience.

**Comment by Trenly** [comment, marked as answer]
WinGet uses the values in the registry that tie back to the `Add or Remove Programs` list in control panel, or `Apps > Installed apps` in Windows Settings.  Correlation is performed based on the values in the DisplayName and Publisher. The ProductCode (key containing the individual values) is sometimes also used. Ensuring the values for these keys are written and match what is in the manifest will ensure that most software is detected properly.

In the case of pre-release versions or beta versions, the most common reason that WinGet may not have proper detection is that typical release strategies use the *exact same* DisplayName, Publisher, and ProductCode. This means that WinGet will attempt the correlation and see that the pre-release version is an exact match for the stable version and map to that. The easiest fix is to change both the DisplayName and ProductCode, or (at very minimum) just the DisplayName

**Reply by marticliment** [reply]
Thanks!
